python/ray/_raylet.pyx

```diff
@@ -1722,6 +1732,25 @@ cdef class CoreWorker:
             self.job_config.ParseFromString(c_job_config.SerializeAsString())
         return self.job_config
 
+    def prepare_runtime_env(self, runtime_env_dict: dict) -> str:
+        """Update parent's runtime env with new env via a simple dict update.
+
+        If the resulting runtime env is empty, fall back to the runtime env
+        set in the JobConfig.  Returns the JSON-serialized runtime env.
+        """
+
+        result_dict = self.get_current_runtime_env_dict()
+        result_dict.update(runtime_env_dict)
```

Comment:
oh I think this might actually be a bug -- we're updating the `current_runtime_env_dict` in-place so future calls will see the changes made here